### PR TITLE
Add MaskedImageModelingOutput 

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
- Adds `MaskedImageModelingOutput` and `TFMaskedImageModelingOutput` classes for masked image modeling / completion / in-painting models.
- Replaces the inaccurate MaskedLMOutput used for ViT and DeiT MIM heads with the new output class
- Ensures backward compatibility by adding `logits` as a property to the new output class

## Before submitting
- [X ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding! 

Let's wait for reviews from @sgugger and @NielsRogge before merging to see if there's anything else that should be added.

File: src/transformers/modeling_outputs.py
Comment (sgugger) [inline src/transformers/modeling_outputs.py:1654]: Maybe issue a warning here that this attribute is deprecated? (and same on the TF class)

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Comment by alaradirik** [comment]
Pinging @NielsRogge for the final approval

File: src/transformers/modeling_outputs.py
Comment (NielsRogge) [inline src/transformers/modeling_outputs.py:1632]:
```suggestion
    Base class for outputs of masked image completion / in-painting models.
    
    Args:
```

File: src/transformers/modeling_tf_outputs.py
Comment (NielsRogge) [inline src/transformers/modeling_tf_outputs.py:959]:
```suggestion
    Base class for outputs of masked image completion / in-painting models.
    
    Args:
```

File: src/transformers/models/swin/convert_swin_simmim_to_pytorch.py
Comment (NielsRogge) [inline src/transformers/models/swin/convert_swin_simmim_to_pytorch.py:165]: Any reason this gets removed?

Comment (alaradirik) [reply src/transformers/models/swin/convert_swin_simmim_to_pytorch.py:165]: Removed accidentally, I'm adding it back

File: src/transformers/modeling_outputs.py
Comment (NielsRogge) [inline src/transformers/modeling_outputs.py:1634]: (nit) models like ViTForMaskedImageModeling don't require `labels` to return a loss, but rather `bool_masked_pos`. So perhaps we can leave out "returned when labels is provided"

**Comment by alaradirik** [comment]
@NielsRogge could you take another look? I think all comments are addressed

File: src/transformers/modeling_outputs.py
Comment (NielsRogge) [inline src/transformers/modeling_outputs.py:1634]: The "bool_masked_pos" argument is highly specific to the xxxForMaskedImageModeling models we have, so I assume this output class is only meant for them?

Otherwise if we plan to use this output class for other models in the future, we may have to remove that in the docstring

Comment (alaradirik) [reply src/transformers/modeling_outputs.py:1634]: The image in-painting / completion models will require bool_masked_pos + the ground truth image (as the input is masked) to compute a loss so I don't think there will be any changes needed, or at least it'll be very minimal

**Pull request merged**
